user_info_helper.go

```diff
@@ -55,3 +55,13 @@ func (x respUserGet) intoUserInfo() UserInfo {
 		QRCodeURL:   x.QRCodeURL,
 	}
 }
+
+func (x respUsersByDept) intoUserInfos() []*UserInfo {
+	var users []*UserInfo
+	for _, user := range x.Users {
+		respUser := respUserGet{UserDetail: *user}
+		userInfo := respUser.intoUserInfo()
```

Comment:
You can refactor a little and move the `intoUserInfo` method to attach to the newly extracted user detail struct instead.